feat: add implementation for TopsortClient (#43)
Topsort/topsort.js: README.md

### Auctions

> To create an auction, first initialize Topsort Client, then use the `createAuction` function. Example:

Reviewer: ```suggestion
To create an auction, first initialize a Topsort Client, then call the `createAuction` function:
```

```js

> import { TopsortClient, TopsortAuction } from "@topsort/sdk";

Reviewer: Can't help but feel there's a lot of "Topsort" everywhere. Do we really need it? We're already in `@Topsort/sdk`.

  // generate your api key in the auction manager - it should look some thing like this
  // note: this is an invalid key and won't work, you need to replace it with your own
  apiKey: "TSE_4S6o1g1CB5tyRENfhDMAn6viR7A5cy3j1JAR",
  userAgent: "Mozilla/5.0", // optional user agent to be added as part of the request

>   timeout: 50 // optional timeout signal to be added asp art of the request

Reviewer: ```suggestion
  timeout: 50 // optional timeout for the request
```

### Events

> To report an event, first initialize Topsort Client, then use the `reportEvent` function. Here is an example:

Reviewer: ```suggestion
To report an event, first initialize a Topsort Client, then call the `reportEvent` function:
```
